### PR TITLE
Add support to configure server port

### LINKED ISSUE #14: Server configuration
Should support configuring which port to listen on by:
- env var
- config file
- flag

precedence (lower number == higher precedence)
1. flag
2. env var
3. Config file

### PR DESCRIPTION
**Pull request by sowmiyamuthuraman**
Fixes #14

### CODE REVIEW

File: igo/igod/server.go [hunk 1]
```diff
@@ -17,7 +17,7 @@ type Server struct {
 }
 
 // Run is
-func (s *Server) Run() {
+func (s *Server) Run(port string) {
```
Comment (beeceej) [inline igo/igod/server.go]: I think it would make more sense if port were a property on the Server struct.

File: igo/igod/server.go [hunk 2]
```diff
@@ -50,8 +50,9 @@ func (s *Server) Run() {
 		}
 		w.Write(b)
 	})
+	fmt.Println(port)
```
Comment (beeceej) [inline igo/igod/server.go]: let's remove this for now, we probably will want more robust diagnostic logs in the future, but that can come later.

File: .config/igo/config.toml [hunk 3]
```diff
@@ -0,0 +1 @@
+port = "9999"
```
Comment (beeceej) [inline .config/igo/config.toml]: the location for this file would be in the users home directory, so I don't think we need to commit this file

File: cmd/igod/main.go [hunk 4]
```diff
@@ -1,17 +1,46 @@
...
+	portPtr := flag.String("port", "", "port on which server should run")
+	flag.Parse()
+	var port string
+	if *portPtr != "" {
+		port = *portPtr
+	} else if os.Getenv("IGO_PORT") != "" {
+		port = os.Getenv("IGO_PORT")
+	} else {
+		var configuration configuration
+		if _, err := toml.DecodeFile(".config/igo/config.toml", &configuration); err != nil {
```
Comment (beeceej) [inline cmd/igod/main.go]: I would use `filepath.Join(os.Getenv("HOME"), ".config", "igo", "config.toml")`.

the location for this file would be in the users home directory which is why I would recommend looking up the users home directory via env var.

**Review by beeceej — CHANGES_REQUESTED** [review]
I noticed a couple things, let me know when you sort those out and I can take another look!

Thanks!

EDIT:
also, don't worry about tests for now. I haven't even spend time setting those up myself (though, those are in the works)

**Comment by sowmiyamuthuraman** [comment]
@beeceej I have done the changes and amended to the same commit.

File: .config/igo/config.toml (hunk 3 above)
Comment (beeceej) [reply .config/igo/config.toml]: @sowmiyamuthuraman looks like this file still stuck around can we remove it it?

**Comment by beeceej** [comment]
@sowmiyamuthuraman let's just remove that config file that is in the repository. 
Once that is handled this looks good

**Review by beeceej — APPROVED** [review]
Looks good! Do you mind if I resolve the go.mod/go.sum conflicts myself?

**Comment by sowmiyamuthuraman** [comment]
> Looks good! Do you mind if I resolve the go.mod/go.sum conflicts myself?

@beeceej I have no problem with it

**Pull request merged**